### DISCUSSION TITLE
Easily convert Vec3 to Vec2

### DISCUSSION THREAD

**Discussion by DasLixou** [bevyengine/bevy#5905] [Q&A] [answered]
Hi! im very new to rust but i must say that bevy is very easy to use and understand - i would say i learned the basics of rust just of 2 hours with bevy :D - but i have one problem. Digging in Vec2 there is no implementation of converting from a Vec3 to a Vec2. So half of my code looks like this: `Vec3::from((transform.translation.x, transform.translation.y, 0.0))`. Is there a simpler way to convert them? thanks :)

**Comment by DJMcNab** [comment, marked as answer]
Why not use [`Vec3::truncate`](https://docs.rs/glam/latest/glam/f32/struct.Vec3.html#method.truncate) to convert from a `Vec3` to a `Vec2`?

The example the code you've given would then use [`Vec2::extend`](https://docs.rs/glam/latest/glam/f32/struct.Vec2.html#method.extend), i.e. `transform.translation.truncate().extend(0.0)`

**Comment by Sheepyhead** [comment]
You can import the swizzle methods from bevy::math::swizzles which lets you do some_vec.xy() or .xz() for example to turn it into a vec2
